modules/broker-auth/src/main/java/org/wso2/broker/auth/user/impl/UserStoreManagerImpl.java

```diff
@@ -0,0 +1,118 @@
+import org.wso2.broker.auth.BrokerAuthException;
+import org.wso2.broker.auth.user.User;
+import org.wso2.broker.auth.user.UserStoreManager;
+import org.wso2.broker.auth.user.UsersFile;
+import org.wso2.carbon.config.ConfigProviderFactory;
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.config.provider.ConfigProvider;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class implements @{@link UserStoreManager} to provide file based user store manager.
+ */
+public class UserStoreManagerImpl implements UserStoreManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            org.wso2.broker.auth.user.impl.UserStoreManagerImpl.class);
+
+    /**
+     * Store the map of users
+     */
+    private static Map<String, User> users = new ConcurrentHashMap<>();
+
+    public Map<String, User> getUsers() {
+        return users;
+    }
+
+    public UserStoreManagerImpl() throws ConfigurationException {
+        Path usersYamlFile;
+        String usersFilePath = System.getProperty(BrokerAuthConstants.SYSTEM_PARAM_USERS_CONFIG);
+        if (usersFilePath == null || usersFilePath.trim().isEmpty()) {
+            // use current path.
+            usersYamlFile = Paths.get("", BrokerAuthConstants.USERS_FILE_NAME).toAbsolutePath();
+        } else {
+            usersYamlFile = Paths.get(usersFilePath).toAbsolutePath();
+        }
+        ConfigProvider configProvider = ConfigProviderFactory.getConfigProvider(usersYamlFile, null);
+        UsersFile usersFile = configProvider
+                .getConfigurationObject(BrokerAuthConstants.USERS_CONFIG_NAMESPACE, UsersFile.class);
+        if (usersFile != null) {
+            List<User> usersList = usersFile.getUsers();
+            for (User user : usersList) {
+                if (user != null && user.getUsername() != null) {
+                    users.put(user.getUsername(), user);
+                } else {
+                    LOGGER.error("User or username can not be null");
+                }
+            }
+        }
+    }
+
+    /**
+     * Authenticate given user with credentials.
+     *
+     * @param userName    userName
+     * @param credentials Credentials
+     * @return Authentication result
+     * @throws BrokerAuthException Exception throws when authentication failed.
+     */
+    @Override
+    public boolean authenticate(String userName, char... credentials) throws BrokerAuthException {
+        User user = users.get(userName);
+        if (user == null) {
+            throw new BrokerAuthException("User not found for the given user name.");
+        } else {
+            if (credentials != null && user.getPassword() != null && Arrays
+                    .equals(credentials, user.getPassword().toCharArray())) {
```

Comment:
I think `getPassword` should return a char array instead of a String for security reasons.